scripts/scil_clean_qbx_clusters.py

```diff
@@ -0,0 +1,285 @@
+
+        if key in ['a', 'r', 'z'] and iterator < len(sft_accepted_on_size):
+            renderer.rm(curr_streamlines_actor)
+            curr_streamlines = sft_accepted_on_size[iterator].streamlines
+            curr_streamlines_actor = actor.line(curr_streamlines,
+                                                opacity=0.8,
+                                                linewidth=clusters_linewidth)
+            renderer.add(curr_streamlines_actor)
+
+        if iterator == len(sft_accepted_on_size):
+            print('No more cluster, press q to exit')
+            renderer.rm(curr_streamlines_actor)
+
+        renwin.Render()
+
+    parser = _build_args_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, args.in_bundles)
+    assert_outputs_exist(parser, args, [args.out_accepted, args.out_rejected])
+
+    if args.out_accepted_dir:
+        assert_output_dirs_exist_and_empty(parser, args,
+                                           args.out_accepted_dir,
+                                           create_dir=True)
+    if args.out_rejected_dir:
+        assert_output_dirs_exist_and_empty(parser, args,
+                                           args.out_rejected_dir,
+                                           create_dir=True)
+
+    if args.verbose:
+        logging.basicConfig(level=logging.INFO)
+
+    if args.min_cluster_size < 1:
+        parser.error('Minimum cluster size must be at least 1.')
+
+    clusters_linewidth = args.clusters_linewidth
+    background_linewidth = args.background_linewidth
+
+    # To accelerate procedure, clusters can be discarded based on size
+    # Concatenation is to give spatial context
+    sft_accepted_on_size, filename_accepted_on_size = [], []
+    sft_rejected_on_size, filename_rejected_on_size = [], []
+    concat_streamlines = []
+    for filename in args.in_bundles:
+        if not is_header_compatible(args.in_bundles[0], filename):
+            return
+        basename = os.path.basename(filename)
+        sft = load_tractogram_with_reference(parser, args, filename,
+                                             bbox_check=False)
+        if len(sft) >= args.min_cluster_size:
+            sft_accepted_on_size.append(sft)
+            filename_accepted_on_size.append(basename)
+            concat_streamlines.extend(sft.streamlines)
+        else:
+            logging.info('File %s has %s streamlines, automatically rejected.',
+                         filename, len(sft))
+            sft_rejected_on_size.append(sft)
+            filename_rejected_on_size.append(basename)
+
+    logging.info('%s clusters to be classified.', len(sft_accepted_on_size))
+    # The clusters are sorted by size for simplicity/efficiency
+    tuple_accepted = zip(*sorted(zip(sft_accepted_on_size,
+                                     filename_accepted_on_size),
+                                 key=get_length_tuple, reverse=True))
```

Comment:
The function get_length_tuple isn't used anywhere else. Consider changing it for a lambda expression

Reply:
I'm not really good with that, if you know exactly how to do it could you help me ?

Reply:
Change the key parameter of the sorted function to : lambda item : len(item[0])